File: search/linear_search/php/linear_search.php
<?php

function linearSearch(array $elements, $number) {
    for($i = 0; $i < count($elements); $i++)
        if($elements[$i] === $number)
        	return $i;
    return -1;
}
function linearSearchTest() {
    $elements = [2,3,5,-12,70,123,256,10,-53,11];
    $num = 10;
    $result = linearSearch($elements, $num);
    
    if ($result === -1) {
        echo "Element not found.";
    } else {
        echo "Element found.";
    }
}
linearSearchTest();

?>

File: search/linear_search/dart/linear_search.dart
void main() {
  test();
}

int linearSearch(List<int> data,int target){
  int targetIndex = -1;
  for(int i = 0;i < data.length;i++){
    if(data[i] == target){
      targetIndex = i;
    }
  }
  return targetIndex;
}

void test(){
  List<int> data = [20,14,6,11,39,9];
  int target = 14;
  int result;
  
  if((result = linearSearch(data,target)) != -1){
    print("$target found at $result");
  }
  else{
    print("$target not found");
  }
}